src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -117,34 +117,36 @@ private TestSuite Build(Assembly assembly, string assemblyNameOrPath, IDictionar
                 {
                     // This cannot be changed without breaking backwards compatibility with old runners.
                     // Deserializes the way old runners understand.
-                    
-                    options.TryGetValue(FrameworkPackageSettings.TestParameters, out object testParameters);
-                    string parametersString = testParameters as string;
-                    if (!string.IsNullOrEmpty(parametersString))
+
+                    if (options.TryGetValue(FrameworkPackageSettings.TestParameters, out var testParameters))
                     {
-                        foreach (string param in parametersString.Split(new[] { ';' }))
+                        var parametersString = (string)testParameters;
```

Comment:
If this can be null, you need a `?` in the type name.
```suggestion
                        var parametersString = (string?)testParameters;
```
Or if the parameter might not be a string at all, keep the original code, but change the type:
```csharp
                        string? parametersString = testParameters as string;
```

Reply:
if it isnt a string, shouldnt we throw an exception?

Reply:
side note. where is this feature documented?

Reply:
The original code didn't throw.

I don't know if it is documented, I couldn't find any setters. Some comments indicate it is for backwards compatibility of old runners.